src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletSupport.scala

```diff
@@ -268,8 +268,18 @@ trait ErgoWalletSupport extends ScorexLogging {
 
     require(inputBoxes.nonEmpty, "There must be at least one input box")
 
+    //filter burnTokens requests
+    val requestsWithoutBurnTokens = requests.filterNot(_.isInstanceOf[BurnTokensRequest])
+    val burnTokens = {
+      val burnRequests = requests.flatMap {
+        case req: BurnTokensRequest => req.assetsToBurn
+        case _ => None
+      }
+      if (burnRequests.isEmpty) Map.empty[ModifierId, Long] else TransactionBuilder.collTokensToMap(burnRequests.toColl)
```

Comment:
@ccellado this would be much cleaner with `val (requestsWithBurnTokens, requestsWithoutBurnTokens) = requests.partition(_.isInstanceOf[BurnTokensRequest])`

Reply:
Just `val (requestsWithBurnTokens, requestsWithoutBurnTokens)` and then : 
```
requestsWithBurnTokens.map(_.asInstanceOf[BurnTokensRequest])
```
I suggested this as your version is filtering the `requests` twice. OCD

Reply:
Now this makes sense. Fixed it, thanks